.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,117 @@
+name: Sub-IoT-Stack CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  run-unit-tests:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Build stack natively
+      run: |
+        mkdir build && cd build
+        platform="NATIVE"
+        cmake ../stack/ -DPLATFORM=$platform -DCMAKE_TOOLCHAIN_FILE="../stack/cmake/toolchains/gcc.cmake" -DBUILD_UNIT_TESTS=y -DFRAMEWORK_CONSOLE_ENABLED=n -DTEST_AES=y -DTEST_FEC=y -DTEST_ALP=y -DTEST_FIFO=y -DMODULE_D7AP=n -DMODULE_ALP_SERIAL_INTERFACE_ENABLED=n
+        make -j
+
+    - name: Run unit tests
+      run: |
+        ./build/tests/aes/test_aes &> results_aes.txt
+        ./build/tests/alp/test_alp &> results_alp.txt
+        ./build/tests/fec/test_fec &> results_fec.txt
+        ./build/tests/fifo/test_fifo &> results_fifo.txt
+
+    - name: Upload the results
+      uses: actions/upload-artifact@v2
+      with:
+        name: results
+        path: |
+          results_aes.txt
+          results_alp.txt
+          results_fec.txt
+          results_fifo.txt
+
+    - name: Handle results
+      run: |
+        if ! grep -q 'AES all unit tests OK !' "results_aes.txt"; then exit 1;  fi
+        if ! grep -q 'Unit-tests for ALP completed' "results_alp.txt"; then exit 1;  fi
+        if ! grep -q 'Input was decoded successfully' "results_fec.txt"; then exit 1;  fi
+        if ! grep -q 'All FIFO tests passed!' "results_fifo.txt"; then exit 1;  fi   
+
+  generate-builds:
+    name: Generate Builds
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up compiler
+      uses: fiam/arm-none-eabi-gcc@v1
+      with:
+        release: '8-2018-q4'
+
+    - name: Build stack
+      run: |
+        mkdir build && cd build
+        platform="B_L072Z_LRWAN1"
+        cmake ../stack/ -DAPP_GATEWAY=y -DAPP_MODEM=y -DAPP_SENSOR_PUSH=y -DPLATFORM=$platform -DFRAMEWORK_DEBUG_ASSERT_REBOOT=y -DMODULE_D7AP_FS_DISABLE_PERMISSIONS=y -DAPP_MODEM_FORWARD_ALP_OVER_SERIAL=y
+        make -j
+        cd ..
+
+    - name: Upload the builds
+      uses: actions/upload-artifact@v2
+      with:
+        name: build
+        path: |
+          build/apps/gateway/
+          build/apps/modem/
+          build/apps/sensor_push/
+
+  run-testsuite:
+    name: Run Testsuite
+    runs-on: self-hosted
+    needs: generate-builds
+
+    steps:
+    - uses: actions/download-artifact@v3
+
+    - name: Flash builds to devices
+      run: |
+        if [ lsof -i -P -n | grep -q JLinkExe ] ; then exit 1; fi
+        if [ lsof -n /dev/ttyUSB0 | grep -q /dev/ttyUSB0 ] ; then exit 1; fi
+        if [ lsof -n /dev/ttyUSB1 | grep -q /dev/ttyUSB1 ] ; then exit 1; fi
```

Comment:
also do this test when running the actual testsuite, could be the tests are triggered at the exact same time